contracts/core/Eip7702Support.sol

```diff
@@ -0,0 +1,58 @@
+pragma solidity ^0.8;
+
+import "../interfaces/PackedUserOperation.sol";
+import "../core/UserOperationLib.sol";
+// SPDX-License-Identifier: MIT
+
+// EIP-7702 code prefix. Also, we use this prefix as a marker in the initCode. To specify this account is EIP-7702.
+uint256 constant EIP7702_PREFIX = 0xef0100;
+
+using UserOperationLib for PackedUserOperation;
+
+    //get alternate InitCode (just for hashing) when using EIP-7702
+    function _getEip7702InitCodeOverride(PackedUserOperation calldata userOp) view returns (bytes32) {
+        bytes calldata initCode = userOp.initCode;
+        if (! _isEip7702InitCode(initCode)) {
+            return 0;
+        }
+        address delegate = _getEip7702Delegate(userOp.getSender());
+        if (initCode.length < 20)
+            return keccak256(abi.encodePacked(delegate));
+        else
+            return keccak256(abi.encodePacked(delegate, initCode[20 :]));
+    }
+
+
+    function _isEip7702InitCode(bytes calldata initCode) pure returns (bool) {
```

Comment:
If you write this function without the assembly tricks, how much more gas does it spend? This code is too elaborate for the task of "compare first three bytes".